src/InnerHit/NestedInnerHit.php

```diff
@@ -48,11 +48,13 @@ class NestedInnerHit implements BuilderInterface
      * @param string           $path
      * @param BuilderInterface $query
      */
-    public function __construct($name, $path, BuilderInterface $query)
+    public function __construct($name, $path, BuilderInterface $query = null)
     {
         $this->setName($name);
         $this->setPath($path);
-        $this->setQuery($query);
+        if ($query) {
```

Comment:
Here we have 2 solutions, the if or this one : 
```php
    public function __construct($name, $path, BuilderInterface $query = null)
    {
        $this->setName($name);
        $this->setPath($path);
        $this->setQuery($query);
    }

    /**
     * @param BuilderInterface $query
     */
    public function setQuery(BuilderInterface $query = null) 
   {
```
I prefer the if one to avoid to be allow to setQuery(null) that is more buggy from my point of view

Like you want.